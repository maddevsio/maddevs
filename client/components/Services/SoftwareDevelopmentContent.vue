<template>
  <div class="content-item col-xl-4 col-lg-4">
    <div class="content-item__content-wrapper">
      <div class="content-item__text-content">
        <client-only>
          <UIItemTitle :item-title="title" class="content-item__title" />
          <UIItemSubTitle :item-sub-title="subTitle" class="content-item__sub-title" />
          <UIParagraph
            :paragraph="firstParagraph"
            :class="`content-item__paragraph-${title.toLowerCase()}`"
            class="content-item__paragraph content-item__paragraph-first"
            v-html="firstParagraph"
          />
          <UIParagraph
            :paragraph="secondParagraph"
            class="content-item__paragraph content-item__paragraph-second"
            v-html="secondParagraph"
          />
        </client-only>
      </div>
      <SoftwareDevelopmentIcons :icons="icons" />
    </div>
  </div>
</template>

<script>
import UIParagraph from '@/components/ui/Services/UIParagraph'
import UIItemTitle from '@/components/ui/Services/UIItemTitle'
import UIItemSubTitle from '@/components/ui/Services/UIItemSubTitle'
import SoftwareDevelopmentIcons from '@/components/Services/SoftwareDevelopmentIcons'

export default {
  name: 'SoftwareDevelopmentContent',
  components: {
    UIItemTitle,
    UIItemSubTitle,
    UIParagraph,
    SoftwareDevelopmentIcons,
  },

  props: {
    title: {
      type: String,
      default: '',
    },

    subTitle: {
      type: String,
      default: '',
    },

    firstParagraph: {
      type: String,
      default: '',
    },

    secondParagraph: {
      type: String,
      default: '',
    },

    icons: {
      type: Array,
      default: Function,
    },
  },
}
</script>

<style lang="scss" scoped>
@import '../../assets/styles/vars';

.content-item {
  &__content-wrapper {
    height: calc(100% - 64px);
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: 32px;
    background-color: $bgcolor--white;
    @media screen and (max-width: 400px) {
      height: calc(100% - 56px);
      padding: 32px 20px 24px;
    }
  }

  &__title,
  &__sub-title,
  &__paragraph {
    color: $text-color--black-lighter;
  }

  &__sub-title {
    margin-bottom: 8px;

    @media screen and (min-width: 400px) and (max-width: 1023px) {
      margin-bottom: 13px;
    }
  }

  &__paragraph-first {
    margin-bottom: 30px;
  }

  &__paragraph-second {
    margin-bottom: 22px;

    @media screen and (min-width: 400px) and (max-width: 991px) {
      margin-bottom: 11px;
    }
  }

  &__paragraph-frontend {
    @media screen and (min-width: 992px) and (max-width: 1070px) {
      width: 183px;
      line-height: 25px;
    }
  }
}
</style>
