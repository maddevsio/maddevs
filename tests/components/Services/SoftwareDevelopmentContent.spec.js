import SoftwareDevelopmentContent from '@/components/Services/SoftwareDevelopmentContent'
import { render, screen } from '@testing-library/vue'

const props = {
  type: 'some-type',
  title: 'some-title',
  subtitle: 'sub-title',
  paragraphs: ['first', 'second'],
  icons: [],
}

describe('SoftwareDevelopmentContent component', () => {
  it('should render correctly', () => {
    const { container } = render(SoftwareDevelopmentContent, {
      props,
      stubs: ['ClientOnly'],
    })

    expect(container).toMatchSnapshot()
  })

  it('should correct render title', () => {
    render(SoftwareDevelopmentContent, {
      props,
      stubs: ['ClientOnly'],
    })

    expect(screen.getAllByText(props.title)).toBeTruthy()
    expect(screen.getAllByTestId('test-paragraph')).toHaveLength(props.paragraphs.length)
  })
})
