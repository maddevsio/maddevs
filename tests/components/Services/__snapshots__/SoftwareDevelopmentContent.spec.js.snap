// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoftwareDevelopmentContent component should render correctly 1`] = `
<div>
  <div
    class="content-item col-xl-4 col-lg-4"
  >
    <div
      class="content-item__content-wrapper"
    >
      <div
        class="content-item__text-content"
      >
        <clientonly-stub>
          <h3
            class="item-title content-item__title"
          >
            
          some-title
        
          </h3>
           
          <h4
            class="item-sub-title content-item__sub-title"
          >
            
          sub-title
        
          </h4>
           
          <p
            class="paragraph content-item__paragraph content-item__paragraph-some-type"
            data-testid="test-paragraph"
          >
            first
          </p>
          <p
            class="paragraph content-item__paragraph content-item__paragraph-some-type"
            data-testid="test-paragraph"
          >
            second
          </p>
        </clientonly-stub>
      </div>
       
      <ul
        class="icons-list"
      />
    </div>
  </div>
</div>
`;
